astronomer/providers/dbt/cloud/hooks/dbt.py

```diff
@@ -0,0 +1,127 @@
+        version = provider_info["versions"]
+        headers["User-Agent"] = f"{package_name}-v{version}"
+        headers["Content-Type"] = "application/json"
+        headers["Authorization"] = f"Token {connection.password}"
+        return headers, tenant
+
+    @staticmethod
+    def get_request_url_params(
+        tenant: str, endpoint: str, include_related: Optional[List[str]] = None
+    ) -> Tuple[str, Dict[str, Any]]:
+        """Form URL from base url and endpoint url"""
+        data: Dict[str, Any] = {}
+        base_url = f"https://{tenant}.getdbt.com/api/v2/accounts/"
+        if include_related:
+            data = {"include_related": include_related}
+        if base_url and not base_url.endswith("/") and endpoint and not endpoint.startswith("/"):
+            url = base_url + "/" + endpoint
+        else:
+            url = (base_url or "") + (endpoint or "")
+        return url, data
+
+    @provide_account_id
+    async def get_job_details(
+        self, run_id: int, account_id: Optional[int] = None, include_related: Optional[List[str]] = None
+    ) -> Any:
+        """
+        Uses Http async call to retrieve metadata for a specific run of a dbt Cloud job.
+
+        :param run_id: The ID of a dbt Cloud job run.
+        :param account_id: Optional. The ID of a dbt Cloud account.
+        :param include_related: Optional. List of related fields to pull with the run.
+            Valid values are "trigger", "job", "repository", and "environment".
+        """
+        endpoint = f"{account_id}/runs/{run_id}/"
+        headers, tenant = await self.get_headers_tenants_from_connection()
+        url, params = self.get_request_url_params(tenant, endpoint, include_related)
+        async with aiohttp.ClientSession(headers=headers) as session:
+            async with session.get(url, params=params) as response:
+                try:
+                    response.raise_for_status()
+                    return await response.json()
+                except ClientResponseError as e:
+                    raise AirflowException(str(e.status) + ":" + e.message)
+
+    async def get_job_status(
+        self, run_id: int, account_id: Optional[int] = None, include_related: Optional[List[str]] = None
+    ) -> int:
+        """
+        Retrieves the status for a specific run of a dbt Cloud job.
+
+        :param run_id: The ID of a dbt Cloud job run.
+        :param account_id: Optional. The ID of a dbt Cloud account.
+        :param include_related: Optional. List of related fields to pull with the run.
+            Valid values are "trigger", "job", "repository", and "environment".
+        """
+        try:
+            self.log.info("Getting the status of job run %s.", str(run_id))
+            response = await self.get_job_details(
+                run_id, account_id=account_id, include_related=include_related
+            )
+            job_run_status: int = response["data"]["status"]
+            return job_run_status
```

Comment:
what about return response["data"]["status"] ?? mypy failing ??

Reply:
Yes mypy is failing so I cast it as int and the function is expecting int